cu29_derive/src/lib.rs

```diff
@@ -397,15 +401,19 @@ fn extract_tasks_types(copper_config: &CuConfig) -> (Vec<String>, Vec<Type>) {
     (all_types_names, all_types)
 }
 
-fn extract_msg_types(runtime_plan: &Vec<CuExecutionStep>) -> Vec<Type> {
+fn extract_msg_types(runtime_plan: &CuExecutionLoop) -> Vec<Type> {
     runtime_plan
+        .steps
         .iter()
-        .filter_map(|step| {
-            if step.output_msg_type.is_none() {
-                None
-            } else {
-                Some(parse_str::<Type>(step.output_msg_type.clone().unwrap().as_str()).unwrap())
+        .filter_map(|unit| match unit {
+            CuExecutionUnit::Step(step) => {
+                if step.output_msg_type.is_none() {
+                    None
+                } else {
+                    Some(parse_str::<Type>(step.output_msg_type.clone().unwrap().as_str()).unwrap())
+                }
```

Comment:
we can use either `if let` or `match` here instead of using unwrap on `step.output_msg_type`

Reply:
oh yeah, this is ugly let me fix this one :) Thanks